### PR TITLE
[receiver/azuremonitor] Add support for azure workload identity auth

### LINKED ISSUE #24451: [receiver/azuremonitorreceiver] Add support for authenticating using AD workload identity
### Component(s)

receiver/azuremonitor

### Is your feature request related to a problem? Please describe.

Currently, the receiver can only authenticate using an azure service principal.

### Describe the solution you'd like

Using managed identity and azure workload identity would be more secure

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cmergenthaler**
**Description:** 
Enhances azuremonitor receiver authentication by using [azure workload identity](https://azure.github.io/azure-workload-identity/docs/). I have added a new config parameter `auth` to specify the used authentication method (default set to service-principal) in order to possibly add more methods in the future.

**Link to tracking Issue:** Fixes #24451 

**Testing:** Tested on my AKS

**Documentation:** Added to README.md

### CODE REVIEW

**Review by fatsheep9146 — COMMENTED** [review]
since the service_principal and workload_identity are both used in multiple places, how about convert them into consts?

**Comment by cmergenthaler** [comment]
> since the service_principal and workload_identity are both used in multiple places, how about convert them into consts?

Good point! I have converted them to be consts

**Review by fatsheep9146 — APPROVED** [review]

**Review by altuner — CHANGES_REQUESTED** [review]
Good job with improving the receiver, I've added few small comments

File: receiver/azuremonitorreceiver/README.md [hunk 1]
```diff
@@ -18,20 +18,29 @@ This receiver scrapes Azure Monitor API for resources metrics.
...
 - `initial_delay` (default = `1s`): defines how long this receiver waits before starting.
 
-### Example Configuration
+Authenticating using service principal requires following settings:
+- `tenant_id`
+- `client_id`
+- `client_secret`
+
+Authenticating using workload identities requires following environment variables being set (automatically by workload identity webhook):
+- `AZURE_TENANT_ID`
```
[the remark this answers is not in the document]
Comment (cmergenthaler) [reply receiver/azuremonitorreceiver/README.md]: Thanks for the hint, I have adjusted to inputs

File: receiver/azuremonitorreceiver/scraper.go [hunk 2]
```diff
@@ -135,8 +137,17 @@ func (s *azureScraper) GetMetricsValuesClient() MetricsValuesClient {
 }
 
 func (s *azureScraper) start(_ context.Context, _ component.Host) (err error) {
-	s.cred, err = s.azIDCredentialsFunc(s.cfg.TenantID, s.cfg.ClientID, s.cfg.ClientSecret, nil)
-	if err != nil {
+	switch s.cfg.Authentication {
```
[the remark this answers is not in the document]
Comment (cmergenthaler) [reply receiver/azuremonitorreceiver/scraper.go]: Done

**Comment by fatsheep9146** [comment]
@cmergenthaler please fix the failed test, thanks

**Comment by cmergenthaler** [comment]
> @cmergenthaler please fix the failed test, thanks

@fatsheep9146 done

**Review by altuner — APPROVED** [review]

**Comment by cmergenthaler** [comment]
Anything missing here? Would be nice to have this merged so that I don't have to rebase and resolve conflicts again.

File: receiver/azuremonitorreceiver/config.go [hunk 3]
```diff
@@ -235,22 +239,44 @@ type Config struct {
 	AppendTagsAsAttributes                  bool                          `mapstructure:"append_tags_as_attributes"`
 }
 
+const (
+	ServicePrincipal = "service_principal"
+	WorkloadIdentity = "workload_identity"
```
Comment (codeboten) [inline receiver/azuremonitorreceiver/config.go:244]: These don't need to be public

```suggestion
	servicePrincipal = "service_principal"
	workloadIdentity = "workload_identity"
```

**Review by codeboten — COMMENTED** [review]
Thanks for the contribution, just a comment re. constants that don't need to be public. Otherwise it looks good

File: receiver/azuremonitorreceiver/config.go (hunk 3 above)
Comment (cmergenthaler) [reply receiver/azuremonitorreceiver/config.go]: thanks for the hint, updated the consts

**Review by codeboten — COMMENTED** [review]
@cmergenthaler please resolve the last conflict and we can get this merged

**Comment by cmergenthaler** [comment]
> @cmergenthaler please resolve the last conflict and we can get this merged

@codeboten done, please merge

**Review by codeboten — APPROVED** [review]

**Comment by codeboten** [comment]
Thanks @cmergenthaler!

**Pull request merged**